src/connection/ConnectionManager.ts

```diff
@@ -54,7 +53,7 @@ export class ConnectionManager {
 
             // if connection is registered and its not closed then throw an error
             if (existConnection.isConnected)
-                throw new AlreadyHasActiveConnectionError(options.name || "default");
+                return existConnection;
```

Comment:
hey, you cannot simply take and change exist logic this way. This change is out of scope of what you are doing, please remove it.